Chemistry/GetGateCount/GetGateCount.cs

```diff
@@ -0,0 +1,270 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// This loads a Hamiltonian from file and performs gate estimates of a
+// - Jordan–Wigner Trotter step
+// - Jordan–Wigner Qubitization iterate
+
+#region Using Statements
+// We will need several different libraries in this sample.
+// Here, we expose these libraries to our program using the
+// C# "using" statement, similar to the Q# "open" statement.
+
+// We will use the data model implemented by the Quantum Development Kit Chemistry
+// Libraries. This model defines what a fermionic Hamiltonian is, and how to
+// represent Hamiltonians on disk.
+using Microsoft.Quantum.Chemistry;
+using Microsoft.Quantum.Chemistry.OrbitalIntegrals;
+using Microsoft.Quantum.Chemistry.JordanWigner;
+using Microsoft.Quantum.Chemistry.Fermion;
+using Microsoft.Quantum.Chemistry.QSharpFormat;
+
+// To count gates, we'll use the trace simulator provided with
+// the Quantum Development Kit.
+using Microsoft.Quantum.Simulation.Simulators.QCTraceSimulators;
+
+// The System namespace provides a number of useful built-in
+// types and methods that we'll use throughout this sample.
+using System.Linq;
+
+// The System.Diagnostics namespace provides us with the
+// Stopwatch class, which is quite useful for measuring
+// how long each gate counting run takes.
+using System.Diagnostics;
+
+// The System.Collections.Generic library provides many different
+// utilities for working with collections such as lists and dictionaries.
+using System.Collections.Generic;
+
+
+// Finally, we use the Mono.Options and System.Management.Automation
+// libraries to make it easy to use this sample from the command line.
+using System.Threading.Tasks;
+#endregion
+
+namespace Microsoft.Quantum.Chemistry.Samples
+{
+
+    // We begin by specifying a data structure that we can use to hold
+    // results from gate counting a particular method on a particular
+    // integral data set.
+    public class GateCountResults
+    {
+        public string IntegralDataPath;
+        public string Method;
+        public double SpinOrbitals;
+        public double TCount;
+        public double RotationsCount;
+        public double CNOTCount;
+        public double NormMultipler;
+        public long ElapsedMilliseconds;
+        public Dictionary<string, string> TraceSimulationStats;
```

Comment:
Should we keep these? I don't see them used or printed.

Reply:
They are in principle accessible via the PowerShell interface, but they're not otherwise used, no.